code-challenges/401challenges/src/main/java/challenges/tree/BinarySearchTree.java

```diff
@@ -0,0 +1,63 @@
+package challenges.tree;
+
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class BinarySearchTree extends BinaryTree{
+    private Node root;
+
+    public Node getRoot() {
+        return root;
+    }
+
+
+    public BinarySearchTree(){ this.root = null; }
+    public BinarySearchTree(int value) {
+        this.root = new Node(value);
+    }
+
+    private Node addValue(Node node, int value){
+      if(node == null){
+          return new Node(value);
+      }
+      if(value < (int)node.getValue()){
+          node.setLeft(this.addValue(node.getLeft(), value));
+      } else if(value > (int)node.getValue()) {
+          node.setRight(this.addValue(node.getRight(), value));
+      } else {
+          return node;
+      }
+      return node;
+    }
+
+    public void add(int value){
```

Comment:
Since developer read code top-down, you should consider defining helper function after you define the method that uses the helper. In this case, `add()` should be defined before `addValue()`